packages/ui/src/routes/+page.svelte

```diff
@@ -10,13 +10,59 @@
 	import { posts } from '$lib/stores/post'
 	import { profile } from '$lib/stores/profile'
 	import { goto } from '$app/navigation'
+	import { browser } from '$app/environment';
 	import Masonry from '$lib/masonry.svelte'
 
-	let y: number
+	let width: number = browser ? window.innerWidth : 0
+	let newColWidth: string
+
+	function checkWidth(w: number) {
+        let val = '100%'
+
+		if (w < 739) {
+			val = '100%'
+		}
+
+		else if (w < 1060) {
+			val = 'minmax(min(100%/2, max(320px, 100%/2)), 1fr)'
+		}
+
+		else if (w < 1381) {
+			val = 'minmax(min(100%/3, max(320px, 100%/3)), 1fr)'
+		}
+
+		else if (w < 1702) {
+			val = 'minmax(min(100%/4, max(320px, 100%/4)), 1fr)'
+		}
+
+		else if (w < 2023) {
+			val = 'minmax(min(100%/5, max(320px, 100%/5)), 1fr)'
+		}
+
+		else if (w < 2560) {
+			val = 'minmax(min(100%/6, max(320px, 100%/6)), 1fr)'
+		}
+
+		else if (w < 3009) {
+			val = 'minmax(min(100%/7, max(320px, 100%/7)), 1fr)'
+		}
+
+		else {
+			val = 'minmax(323px, 1fr)'			
+		}
+
+        return val
+    }
```

Comment:
Better pattern here is to return the value after each if. It makes it much more readable and compact :). So something like:
```ts
	function checkWidth(w: number) {
 		if (w < 739) {
 			return '100%'
 		}
 		if (w < 1060) {
 			return = 'minmax(min(100%/2, max(320px, 100%/2)), 1fr)'
 		}

 		// ...

 	 if (w < 2560) {
 			return 'minmax(min(100%/6, max(320px, 100%/6)), 1fr)'
 		}
 	 if (w < 3009) {
 			return 'minmax(min(100%/7, max(320px, 100%/7)), 1fr)'
 		}
 		return 'minmax(323px, 1fr)'			
}
```

Reply:
I know Filoozom will not like me for this, but you can also do something like this:
```ts
function checkWidth(w: number) {
 		if (w < 739) return '100%'
 		if (w < 1060) return = 'minmax(min(100%/2, max(320px, 100%/2)), 1fr)'

 		// ...

 		if (w < 2560) return 'minmax(min(100%/6, max(320px, 100%/6)), 1fr)'
 		if (w < 3009) return 'minmax(min(100%/7, max(320px, 100%/7)), 1fr)'
 		return 'minmax(323px, 1fr)'			
}
```